consoles/sshVirtsh.pm

```diff
@@ -270,26 +270,9 @@ sub add_vnc {
 sub add_serial_console {
     my ($self, $args) = @_;
 
-    my $doc     = $self->{domainxml};
-    my $devices = $self->{devices_element};
-    my $port    = $args->{port} // '1';
-
-    my $serial = $doc->createElement('serial');
-    $serial->setAttribute(type => 'pty');
-
-    # set the port number
-    my $target = $doc->createElement('target');
-    $target->setAttribute(port => $port);
-    $serial->appendChild($target);
-
-    # set the name of the serial port used to refer to it when calling 'virsh console'
-    # note: This doesn't seem to have any effect, but set it in accordance with the 'default' we expect to get.
-    my $alias = $doc->createElement('alias');
-    $alias->setAttribute(name => 'serial' . $port);
-    $serial->appendChild($alias);
-
-    $devices->appendChild($serial);
-    return;
+    my $port    = $args->{port}    // '1';
+    my $pty_dev = $args->{pty_dev} // 'serial';
+    $self->add_pty({pty_dev => $pty_dev, pty_dev_type => 'pty', target_type => 'virtio', target_port => $port});
```

Comment:
It is nice that you remove the redundant code here. This throws away the code for setting the alias but it hadn't any effect anyways (at least in my tests).

---

You now set the target type to `virtio`. Did you do this on purpose? I think that virtio is qemu specific and would maybe break Hyper-V and Xen. When I remember correctly, @coolo insistent that I don't use virtio when adding the serial console.

Reply:
You're right. Virtio is going to be used in s390x, but it does not have to be specified (it's the default there), so I'll remove it.

Reply:
Good, and I see that you've already changed the code accordingly.